src/engine/GameState.java

```diff
@@ -205,4 +206,12 @@ public final void setLivesRecovery() {
 	public final int getBulletsRemaining() {
 		return BulletsRemaining;
 	}
+	public Color getPlayerColor(){
```

Comment:
[P2]

This `getPlayerColor()` method is a bit structurally complex.

Essentially, a getter is used to get the state of an entity, or more broadly, to get information about an object.

However, the current `getPlayerColor()' method acts like a createColor, which is different from a setter and a getter in that it creates a value of color, checks it, and returns the value.

This adds complexity and makes for a slightly odd structure where you have to call `getPlayerColor` to impose a value on a ship from `setPlayerColor`. I'm not sure where the create function belongs and if it's doing two things within one method, it might be a good idea to separate them in the future.

--

이 `getPlayerColor()` 매서드는 조금 구조적으로 복잡하게 짜였습니다.

본디 getter라 함은 해당 entity의 상태를 가져올때, 넓게는 대상에 대한 정보를 얻고 싶을때 씁니다.

하지만 현재의 `getPlayerColor()` 매서드는 color의 값을 만들어 내고 검사후 값을 반환하는 즉 setter의 역할이나 getter의 성격과는 다른 createColor와 같은 역할을 하고 있습니다.

이로 인해 복잡도가 증가하고 `setPlayerColor에`서 ship에 값을 부과하기 위해 `getPlayerColor`를 불러야 하는 조금 이상한 구조를 띄고 있습니다. 만드는 기능이 어디에 속해야 하고 한 매서드 내에서 2가지의 기능을 할 경우 분리를 해보는 게 앞으로는 좋을 것 같습니다.